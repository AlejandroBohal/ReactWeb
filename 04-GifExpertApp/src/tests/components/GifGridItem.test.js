import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import React from 'react';
import GifGridItem from '../../components/GifGridItem';

describe('Testing GifExpertApp component', () => {
    const title = 'test title';
    const url = 'https://localhost/test';
    let wrapper = shallow(<GifGridItem title={title} url={url} />);
    beforeEach( () =>{
        wrapper = shallow(<GifGridItem title={title} url={url}/>);    
    });
    test('should load <GifGridItem /> correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
    test('should have a paragraph with title', () => {
        const p = wrapper.find('p').text().trim();
        expect(p).toBe(title);
    });
    test('should have img src', () => {
        const {src,alt} = wrapper.find('img').props();
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });
    test('should have animate class', () => {
        const {className} = wrapper.find('div').props();
        //expect( className.includes('animate__fadeIn')).toBe(true);
        expect(className).toContain('animate__fadeIn');
    });
    
    
    
})