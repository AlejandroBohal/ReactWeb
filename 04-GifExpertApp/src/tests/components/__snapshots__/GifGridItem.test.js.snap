// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifExpertApp component should load <GifGridItem /> correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="test title"
    src="https://localhost/test"
  />
  <p>
    test title
  </p>
</div>
`;
